File: src/main/kotlin/net/yttravelers/spotshare/entities/RouteDetailTourism.kt
package net.yttravelers.spotshare.entities

import com.fasterxml.jackson.annotation.JsonIgnore
import javax.persistence.Table
import javax.persistence.*

@Entity
@Table(name = "routeDetailTourism")
class RouteDetailTourism {
    @Id
    var routeDetailId: Int? = null

//    @JsonIgnore
    @OneToOne
    @JoinColumn(name = "routeDetailId")
    @MapsId
    lateinit var routeDetail: RouteDetail

    @ManyToOne(fetch = FetchType.EAGER)
    lateinit var tourism: Tourism

    var tourismRate: Int? = null

    var tourismMinutes: Int? = null

    var tourismCost: Int? = null
}

File: src/main/angular/src/app/model/route-detail-tourism.ts
import { ITourism } from "./tourism";

/**
 * ルート詳細チェックリスト
 */
export interface IRouteDetailTourism {
  /** ルート詳細ID */
  routeDetailId: string;
  /** 観光地ID */
  tourismId: string;
  /** 行きたい度 */
  tourismRate: string;
  /** 観光時間 */
  tourismMinutes: string;
  /** 観光費用 */
  tourismCost: string;
  /** 観光地 */
  tourism: ITourism;
}